### PR TITLE
Add comments about runner labels

### PR DESCRIPTION
**Pull request by malfet**
To distinguish between org-wide and repo-specific runners as well as highlight where they are hosted (by DevInfra, LF or various partners

Delete unused `bm-runner`

### CODE REVIEW

File: .github/actionlint.yaml [hunk 1]
```diff
@@ -13,16 +16,22 @@ self-hosted-runner:
...
+    # Organization wide AWS Windows runners
     - windows.4xlarge.nonephemeral
     - windows.8xlarge.nvidia.gpu
     - windows.8xlarge.nvidia.gpu.nonephemeral
     - windows.g5.4xlarge.nvidia.gpu
-    - bm-runner
+    # Repo-specific AMD hosted MI300 runners
     - linux.rocm.gpu
+    # Repo-specific Apple hosted M2 runners
+    - macos-m2-metal
```
Comment (huydhn) [inline .github/actionlint.yaml:29]: Do you plan to rename or add the `macos-m2-metal` label here?  I see that it has `macos-m2-14` label atm, and I mistakenly thought that it was a typo with our `macos-m1-14` runner in https://github.com/pytorch/pytorch/pull/127582.  I'm going to put the runner back now with the correct label

**Review by huydhn — APPROVED** [review]

**Comment by huydhn** [comment]
Also the new XPU runners https://github.com/pytorch/pytorch/blob/main/.github/workflows/xpu.yml?

**Comment by malfet** [comment]
> Also the new XPU runners https://github.com/pytorch/pytorch/blob/main/.github/workflows/xpu.yml and aarch64 https://github.com/pytorch/pytorch/blob/main/.github/workflows/linux-aarch64.yml?

Let's do it in follow-up PRs (I wonder if newer actionlint supports composite workflows)

**Pull request closed without merging**